ONSdigital/eq-author-app
# Added firebase jwt token Verification

### What is the context of this PR?
Verification of JWT token against firebase to ensure user is valid

### How to review
**IMPORTANT: Do NOT commit created file .firebase-admin-service-account.json **

1. Generate a private key file for firebase service account and save to eq-author-api/.firebase-admin-service-account.json - see [https://firebase.google.com/docs/admin/setup/](https://firebase.google.com/docs/admin/setup/)
   
   To generate a private key file for your service account:

   1. In the Firebase console, open Settings > Service Accounts.

   2. Click Generate New Private Key, then confirm by clicking Generate Key.

   3. Save the JSON file containing the key to .eq-author-api/.firebase-admin-service-account.json

2. login and use site as normal - ensure login and questionnaire editing work

3. Use dev tools to modify access token (Application > Local Storage > accessToken) with validtoken but invalid user and observe 401 error being returned from requests. Editing of questionnaire is not possible. example jwt token is: `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJodHRwczovL3NlY3VyZXRva2VuLmdvb2dsZS5jb20vb25zLWVxLWF1dGhvci1hcHAiLCJhdWQiOiJvbnMtZXEtYXV0aG9yLWFwcCIsImF1dGhfdGltZSI6MTU2MDI2MTIzOSwidXNlcl9pZCI6ImtaQXZWWHZsRkpnSkx5R0RPajg2cExFQ3FVbDIiLCJzdWIiOiJrWkF2Vlh2bEZKZ0pMeUdET2o4NnBMRUNxVWwyIiwiaWF0IjoxNTYwMjYxMjM5LCJleHAiOjE1NjAyNjQ4MzksImVtYWlsIjoiYW5kcmV3MkBsZW1vbmNvZy5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiZmlyZWJhc2UiOnsiaWRlbnRpdGllcyI6eyJlbWFpbCI6WyJhbmRyZXcyQGxlbW9uY29nLmNvbSJdfSwic2lnbl9pbl9wcm92aWRlciI6InBhc3N3b3JkIn19.mP1he9JpqdOMaL1fbO0KPMSqKY6dq7vtdyBaVZvcYxA`

4. Delete accessToken from localStorage and ensure application doesn't work


### Dependencies
- https://github.com/ONSdigital/eq-author-terraform/pull/18
- https://github.com/ONSdigital/eq-deploy/pull/78
